### PR TITLE
Family Planning importing previous month data

### PR DESCRIPTION
**Pull request by suruchee**

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
+openelis.db.url: jdbc:postgresql://localhost/clinlims?user=clinlims&password=clinlims
+submission.audit.folder: /dhis-integration-data
 log4j.config.file: log4j.properties
 server.port: 8040
 server.context-path: /dhis-integration/
-submission.audit.folder: /dhis-integration-data
-dhis.imam.dataset.id: UTGrrHEFwtt
\ No newline at end of file
+dhis.imam.dataset.id: UTGrrHEFwtt
+dhis.familyPlanning.dataset.id: JOAyT4yOEcs
```
Comment (laxmanm7) [inline src/main/resources/application.yml]: @suruchee  | @dipakthapaofficial, lets put this program specific configuration in the program specific configuration files.

File: src/main/resources/static/js/init.js [hunk 2]
```diff
@@ -158,56 +147,72 @@ function download(index) {
...
-	var endMonth = 3; //Ashadh
-	var programName = element('program-name', index).html();
+    var yearRange = element('fiscal-year', index).val();
+    console.log("yearRange="+yearRange)
+    var years = yearRange.split('-');
+    console.log(years)
+    var startYear = years[0];
+    var startMonth = 4; //Shrawan
+    var endYear = years[1];
+    var endMonth = 3; //Ashadh
```
Comment (laxmanm7) [inline src/main/resources/static/js/init.js]: @dipakthapaofficial  why these months numbers are hardcoded?

File: src/main/java/com/possible/dhis2int/Properties.java [hunk 3]
```diff
@@ -5,43 +5,46 @@
...
+
 	@Value("${submission.audit.folder}")
 	public String submissionAuditFolder;
-	
+
 	@Value("${dhis.imam.dataset.id}")
 	public String dhisImamDataSetId;
+
+	@Value("${dhis.familyPlanning.dataset.id}")
+	public String dhisFamilyPlanningDataSetId;
```
Comment (laxmanm7) [inline src/main/java/com/possible/dhis2int/Properties.java]: @suruchee  User standard formatters as I see lots of format changes in other files as well.

File: src/main/java/com/possible/dhis2int/db/DatabaseDriver.java [hunk 4]
```diff
@@ -173,4 +173,38 @@ public void dropImamTable() throws DHISIntegratorException {
...
+			String insertFamilyPlanningData = new StringBuffer(
+					"INSERT INTO familyPlanning(vasectomy_user, pills_user, other_user,  minilap_user, IUCD_user, implant_user, depo_user, condoms_user) SELECT ")
+					.append(numberOfVasectomyUser).append(", ").append(numberOfPillsUser).append(", ")
+					.append(numberOfOtherUser).append(", ").append(numberOfMinilipUser).append(", ")
+					.append(numberOfIUCDUser).append(", ").append(numberOfImplantUser).append(", ")
+					.append(numberOfDepoUser).append(", ").append(numberOfCondomsUser).toString();
+			statement.executeUpdate(insertFamilyPlanningData);
+
+		} catch (SQLException e) {
+			throw new DHISIntegratorException(String.format("Failed to create table familyPlanning"), e);
```
Comment (laxmanm7) [inline src/main/java/com/possible/dhis2int/db/DatabaseDriver.java:198]: Trace this exception and make sure this exception is not swallowed somewhere as it seems its not handled from its callers, Try replicating the error for this.

File: src/main/java/com/possible/dhis2int/web/DHISIntegrator.java [hunk 5]
```diff
@@ -316,29 +326,29 @@ public FileSystemResource downloadSubmissionLog(HttpServletResponse response) th
 
 	@RequestMapping(path = "/download")
 	public void downloadReport(@RequestParam("name") String name, @RequestParam("year") Integer year,
-			@RequestParam("month") Integer month, @RequestParam("isImam") Boolean isImam, HttpServletResponse response)
+							   @RequestParam("month") Integer month, @RequestParam("isImam") Boolean isImam,@RequestParam("isFamily") Boolean isFamily, HttpServletResponse response)
 			throws JSONException, IOException {
 		ReportDateRange reportDateRange = new DateConverter().getDateRange(year, month);
-		if (isImam != null && isImam) {
```
Comment (laxmanm7) [inline src/main/java/com/possible/dhis2int/web/DHISIntegrator.java]: @suruchee , are these conditions for Imam intentionally deleted ? if so whats solution for downloading Imam reports in other ways?

File: src/main/resources/static/js/init.js [hunk 6]
```diff
@@ -79,46 +69,45 @@ function fiscalYearRange(start, end) {
...
             $("#programs-yearly").html(Mustache.render(template, content));
         });
     });
 }
 
-function getContent(isYearlyReport, canSubmitReport) {
+function getContent(isYearlyReport) {
     return getDHISPrograms().then(function (programs) {
-    	if(isYearlyReport) {
-    		return {years: fiscalYears, programs: programs, isYearlyReport: isYearlyReport, canSubmitReport: canSubmitReport};
```
Comment (laxmanm7) [inline src/main/resources/static/js/init.js:118]: canSubmitReport codes omitted?? I think these are for the privileges checkings!

File: src/main/resources/static/js/init.js [hunk 7]
```diff
@@ -79,46 +69,45 @@ function fiscalYearRange(start, end) {
...
-    
+
     $('[id^="fiscal-year-"]').val((bsDate.bsYear - 1) + '-' + bsDate.bsYear);
 }
 
 function renderPrograms() {
-	return $.get('html/programs.html').then(function (template) {
-		var isYearlyReport= false;
-		var canSubmitReport = hasReportingPrivilege;
-        return getContent(isYearlyReport, canSubmitReport).then(function (content) {
```
Comment (laxmanm7) [inline src/main/resources/static/js/init.js:100]: omitted canSubmitReport ?

**Review by laxmanm7 — CHANGES_REQUESTED** [review]
@suruchee  | @dipakthapaofficial  please go through all the review comments

File: src/main/java/com/possible/dhis2int/web/DHISIntegrator.java (hunk 5 above)
Comment (suruchee) [reply src/main/java/com/possible/dhis2int/web/DHISIntegrator.java]: It was by mistake and I will add it up.

File: src/main/resources/static/js/init.js (hunk 2 above)
Comment (dipakthapaofficial) [reply src/main/resources/static/js/init.js]: Month Value for fiscal year. Ashadh start of fiscal year and shrawan end of fiscal year

**Review by laxmanm7 — APPROVED** [review]

**Review by laxmanm7 — APPROVED** [review]
The requested changes has been aproved

**Pull request merged**